src/test/java/io/appform/dropwizard/actors/actor/metadata/MessageMetadataProviderTest.java

```diff
@@ -0,0 +1,108 @@
+        MockitoAnnotations.openMocks(this);
+
+        List<String> messageMetaGeneratorClasses = Arrays.asList(CustomMessageMetaHeaderGenerator.class.getName());
+        messageMetadataProvider = new MessageMetadataProvider(messageMetaGeneratorClasses);
+    }
+
+    @Test
+    void shouldCreateBlankMessageMetadataWhenMessageMetaGeneratorClassesIsNull() {
+        MessageMetadataProvider messageMetadataProvider = new MessageMetadataProvider(null);
+        MessageMetadata messageMetadata = messageMetadataProvider.createMetadata(mockEnvelope, mockProperties);
+        assertNotNull(messageMetadata);
+
+        assertFalse(messageMetadata.isRedelivered());
+        assertEquals(0, messageMetadata.getDelayInMs());
+        assertFalse(messageMetadata.isExpired());
+
+        assertNull(messageMetadata.getContentType());
+        assertNull(messageMetadata.getContentEncoding());
+        assertNull(messageMetadata.getHeaders());
+        assertNull(messageMetadata.getDeliveryMode());
+        assertNull(messageMetadata.getPriority());
+        assertNull(messageMetadata.getCorrelationId());
+        assertNull(messageMetadata.getReplyTo());
+        assertNull(messageMetadata.getExpiration());
+        assertNull(messageMetadata.getMessageId());
+        assertNull(messageMetadata.getMsgSentTimestamp());
+        assertNull(messageMetadata.getType());
+        assertNull(messageMetadata.getUserId());
+        assertNull(messageMetadata.getAppId());
+        assertNull(messageMetadata.getClusterId());
+    }
+
+    @Test
+    void shouldGenerateMetadataWithCustomMessageMetaHeader() {
+        MessageMetadata messageMetadata = messageMetadataProvider.createMetadata(mockEnvelope, mockProperties);
+
+        assertNotNull(messageMetadata.getHeaders(), "Headers shouldn't be null");
+        assertEquals(CUSTOM_HEADER_VALUE, messageMetadata.getHeaders().get(CUSTOM_HEADER_KEY));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenEnvelopeIsNull() {
+        Exception exception = assertThrows(NullPointerException.class,
+                () -> messageMetadataProvider.createMetadata(null, mockProperties));
+
+        assertEquals("Envelope should not be null", exception.getMessage(),
+                "Exception message should match");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenPropertiesIsNull() {
+        Exception exception = assertThrows(NullPointerException.class,
+                () -> messageMetadataProvider.createMetadata(mockEnvelope, null));
+
+        assertEquals("AMQP.BasicProperties should not be null", exception.getMessage(),
+                "Exception message should match");
+    }
+
+    public static class CustomMessageMetaHeaderGenerator implements MessageMetadataGenerator {
+
+        @Override
+        public void generate(MessageMetaContext messageMetaContext, MessageMetadata messageMetadata) {
+            Map<String, Object> headers = messageMetaContext.getHeaders();
+            if (Objects.isNull(headers)) {
```

Comment:
lol there is zero need to use these functions for null and not null check outside of lambdas. It is just doing one dereferencing and a function call instead of one operator comparison.

Reply:
I have seen this in other places as well please remove the usage of this functions in procedural code